pkg/k8shandler/desirednodestate.go

```diff
@@ -478,6 +487,38 @@ func (cfg *desiredNodeState) getESContainer() v1.Container {
 	}
 }
 
+func (cfg *desiredNodeState) getProxyContainer() v1.Container {
```

Comment:
Nit, but can we start using golang conventions for initializing structs:  "[new|New]ProxyContainer" depending on the visibility

Reply:
I'm simply continuing in the pattern that's already all over the code base. It might not be as idiomatic, but it's consistent.
But I will revisit this in 4.1.